tests/unit/LSYiiValidatorsTest.php

```diff
@@ -27,4 +27,155 @@ public function testHtmlEntityDecodeFilter()
             $this->assertEquals($expected, $actual);
         }
     }
+
+    /**
+     * Testing that the xssfilter attribute will always be
+     * false for super admin users even if filterxsshtml is
+     * changed.
+     */
+    public function testXssFilterAttributeForSuperAdmin()
+    {
+        //Mocking super admin login.
+        \Yii::app()->session['loginID'] = 1;
+        $superAdminValidator = new \LSYii_Validators();
+
+        $this->assertTrue(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml should be true by default.');
+        $this->assertFalse($superAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Changing filterxsshtml.
+        \Yii::app()->setConfig('filterxsshtml', false);
+        $newSuperAdminValidator = new \LSYii_Validators();
+
+        $this->assertFalse(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml was just changed to false.');
+        $this->assertFalse($newSuperAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+    }
+
+    /**
+     * Testing that the xssfilter attribute varies
+     * for regular users depending on filterxsshtml.
+     */
+    public function testXssFilterAttributeForRegularUsers()
+    {
+        //Create user.
+        $newPassword = createPassword();
+        $userId = \User::insertUser('test_user', $newPassword, 'John Doe', 1, 'jd@mail.com');
+
+        //Mocking regular user login.
+        \Yii::app()->session['loginID'] = $userId;
+        $regularUserValidator = new \LSYii_Validators();
+
+        $this->assertTrue(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml should be true by default.');
```

Comment:
Then we can not use this test on a local dev instance :(

Reply:
@olleharstedt on some local instance : we can not be sure to have filterxsshtml to ON by default.

Did we test default options ?

Reply:
I'm unsure here …

I use an existing system to create test, but manual "Important.png  Warning : NEVER run tests on a production system. The tests WILL delete and modify the limesurvey database as well as the config.php file. Always do a backup of important data before running the tests."

https://manual.limesurvey.org/index.php?title=Regression_and_unit_tests

Reply:
Yes : but when construct test : i have my own test DB ;)

Not really important